qless-java/src/main/java/com/moz/qless/Queue.java

```diff
@@ -23,6 +22,91 @@
 import redis.clients.jedis.Transaction;
 
 public class Queue {
+
+  public static class JobSpec {
```

Comment:
I think you already have all of the setters. You would just need the
getters. Also remember that in Java `protected` implies package scope, so
you could actually restrict field access by using the default access.
`protected` means package + derived classes, which the derived classes can
be in *any* package. Since this class should probably be `final`,
`protected` is meaningless.

-- 
Brandon Forehand
On Feb 9, 2015 7:07 AM, "Dan Lecocq" <notifications@github.com> wrote:

> In qless-java/src/main/java/com/moz/qless/Queue.java
> <https://github.com/seomoz/qless-java/pull/21#discussion_r24333283>:
>
> > @@ -23,6 +22,91 @@
> >  import redis.clients.jedis.Transaction;
> >
> >  public class Queue {
> > +
> > +  public static class JobSpec {
>
> Yeah, it's a bit of a stretch. I had wanted to avoid making all the
> getters and setters, but I guess the alternative would be making all these
> members package-protected?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/seomoz/qless-java/pull/21/files#r24333283>.
>